### ISSUE TITLE
No clear use for <button part="button">

### ISSUE THREAD

**Issue by Seanmclem** [PolymerLabs/lit-element-starter-ts#26]
Location: src/my-element.ts ... Line 49
There is `<button @click=${this._onClick} part="button">`

I can't find documentation from any sources about this `part` attribute. The file in the JSDoc references `@csspart button - The button` but makes no attempt to explain what that means or how it relates to css parts. Documentation on css parts does not mention a `part` attribute.

**Comment by hcgaron** [comment]
I agree with the original issue, although my understanding is a bit different.

I can see how the part attribute works in html / css, but I can't seem to get it working inside the `static styles` ttl.

For example, inside the styles ttl if I do:

```
::part(button) {
      background-color: yellow;
      padding: 3rem;
    }
```
It has no effect.  I've tried typical "scss" style selectors like adding `&::part(button)` and even `:host::part(button)` (as well as every variant, such as `:host(*) my-element::part(button)` etc... to try to target the part attribute in question but no luck.

My assumption is that this is only accessible from an external style sheet or parent element?  If someone can give a minimal example that would be helpful.
